lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -290,6 +290,8 @@ fetchFromRemote
     -> IO (Maybe StakePoolMetadata)
 fetchFromRemote tr builders manager pid url hash = runExceptTLog $ do
     chunk <- getChunk `fromFirst` builders
+    when (BS.length chunk >= 513) $ throwE
+        "Metadata exceeds max length of 512 bytes"
```

Comment:
Why 513 ?

Reply:
Because 512 is the max allowed length. By peeking one byte further, we're able to determine whether this property holds.

If you mean to use `(BS.length chunk > 512)` instead, then sure, also can.

Reply:
> If you mean to use (BS.length chunk > 512) instead, then sure, also can.

Well, it's not much this one that bothers me (although `> 512` reflects better the actual property than `>= 513`), but it's the `brReadSome body 513` below.

Reply:
Yes, that is on purpose. We no longer cut off the body, but fail for anything that's longer than `512` bytes, so we must *at least* read 513 bytes. If the body is 512 bytes, it will only be 512, there should be no padding.

Reply:
> fail for anything that's longer than 512 bytes, so we must at least read 513 bytes. 

I do not understand how you draw this conclusion from this hypothesis? :no_mouth:

Reply:
clarified in DM. My bad, we indeed want to know whether the body is longer than 512 bytes and can't do so if only reading up to 512 bytes :+1:

Reply:
Note that I still think that this inequality would be better written as `BS.length chunk > 512` to reflect the actual predicate.